sysfs/class_infiniband_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+package sysfs
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestParseSlowRate(t *testing.T) {
+	rate, err := parseRate("2.5 Gb/sec (1X SDR)")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if rate != 312500000 {
+		t.Errorf("Result for InfiniBand rate not correct: want %v, have %v", 312500000, rate)
+	}
+}
+
+func TestParseRate(t *testing.T) {
+	rate, err := parseRate("500 Gb/sec (4X HDR)")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if rate != 62500000000 {
+		t.Errorf("Result for InfiniBand rate not correct: want %v, have %v", 62500000000, rate)
+	}
+}
+
+func TestInfiniBandClass(t *testing.T) {
+	fs, err := NewFS(sysTestFixtures)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	got, err := fs.InfiniBandClass()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var (
+		port1LinkDowned               uint64 = 0
```

Comment:
Nit: omit any `= 0`; it's inferred automatically.

Reply:
Done, but I like to prefer the explicit definition.